smif/sample_project/models/energy_demand.py

```diff
@@ -11,27 +12,62 @@ def initialise(self, initial_conditions):
         pass
 
     def simulate(self, data):
-        # receive population, energy_demand at annual/national
-        self.logger.info("EDMWrapper received inputs in %s", data.current_timestep)
+
+        # Get the current timestep
+
+        now = data.current_timestep
+        self.logger.info("EDMWrapper received inputs in %s",
+                         now)
+
+        # Demonstrates how to get the value for a model parameter
+        parameter_value = data.get_parameter('smart_meter_savings')
+        self.logger.info('Smart meter savings: %s', parameter_value)
+
+        # Demonstrates how to get the value for a model input
+        # (defaults to the current time period)
+        current_energy_demand = data.get_data('energy_demand')
+        self.logger.info("Current energy demand in %s is %s",
+                         now, current_energy_demand)
+
+        # Demonstrates how to get the value for a model input from the base
+        # timeperiod
+        base_year_enum = RelativeTimestep.BASE
```

Comment:
Suggest not bothering with the local variable `base_year_enum` here (and `prev_...` below) and use `RelativeTimestep.BASE` directly.

Reply:
Does this naming feel ergonomic? Could do `Timestep.BASE_YEAR`, `data.get_base_year()`, `data.base_timestep` (to match `data.current_timestep`) ...